src/components/ReferenceViewComponent.js

```diff
@@ -122,44 +137,75 @@ class ReferenceViewComponent {
    * @private
    */
   #renderPdfReference = (pageNumber) => {
-    this.components.pdfPageNumber = pageNumber;
+    //this.components.pdfPageNumber = pageNumber;
     this.components.sideNav.className = "no-width";
-    this.components.sidePageReferenceContainer.setAttribute(
-      "id",
-      "side-page-reference-container"
-    );
-    this.components.sidePageReferenceContainer.setAttribute(
-      "class",
-      "canvas__container"
-    );
+    this.components.sidePageReferenceContainer.setAttribute("id", "side-page-reference-container");
+    //this.components.sidePageReferenceContainer.setAttribute("class", "canvas__container");
 
     if (this.pdfDoc === null) {
       throw new Error("PDFDocument object missed");
     }
-    this.pdfDoc
-      .getPage(this.components.pdfPageNumber)
-      .then((page) => {
-        const ctxReference =
-          this.components.sidePageReferenceContainer.getContext("2d");
-        this.components.viewport = page.getViewport({
-          scale: 1,
-        });
-
-        this.components.sidePageReferenceContainer.height =
-          this.components.viewport.height;
-        this.components.sidePageReferenceContainer.width =
-          this.components.viewport.width;
-
-        const renderCtx = {
-          canvasContext: ctxReference,
-          viewport: this.components.viewport,
-        };
-        page.render(renderCtx);
-      })
-      .catch((err) => {
-        console.log(err.message); // TODO: handle error in some way
-      });
+
+    this.#renderPageReference(
+      this.pdfDoc,
+      this.components,
+      pageNumber
+    );
+    
   };
+
+  /**
+ * Function to render the page of the reference
+ * @param {pdfDoc} pdfDoc PDF document
+ * @param {Object} component object that holds DOM elements that are within component
+ * @param {HTMLElement} pageNumber number of the page of the reference selected
+ */
+#renderPageReference = (pdfDoc, component, pageNumber) => {
```

Comment:
Like I said on discord, content of this method is repeated so many times in our code, so it should be extracted in some kind of service and code repetition should be eliminated... If you manage to do this that would be great.

Reply:
Done